pkg/controllers/federatedcluster/util.go

```diff
@@ -54,33 +54,34 @@ func setClusterCondition(
 func getNewClusterOfflineCondition(
 	status corev1.ConditionStatus,
 	conditionTime metav1.Time,
+	oldCondition *fedcorev1a1.ClusterCondition,
 ) fedcorev1a1.ClusterCondition {
 	condition := fedcorev1a1.ClusterCondition{
 		Type:               fedcorev1a1.ClusterOffline,
+		Status:             status,
 		LastProbeTime:      conditionTime,
 		LastTransitionTime: conditionTime,
 	}
+	if oldCondition != nil && oldCondition.Status == status {
+		condition.LastTransitionTime = oldCondition.LastTransitionTime
```

Comment:
how about we do this in `setClusterCondition` instead of repeating it for each `getNewXXXCondition` function definition?

Reply:
fixed, ptal cc. @mrlihanbo